include/envoy/http/filter.h

```diff
@@ -102,14 +102,18 @@ class StreamFilterCallbacks {
    * caching where applicable to avoid multiple lookups.
    *
    * NOTE: This breaks down a bit if the caller knows it has modified something that would affect
-   *       routing (such as the request headers). In practice, we don't do this anywhere currently,
-   *       but in the future we might need to provide the ability to clear the cache if a filter
-   *       knows that it has modified the headers in a way that would affect routing. In the future
-   *       we may also want to allow the filter to override the route entry.
+   *       routing (such as the request headers). In the future we may also want to allow the
+   *       filter to override the route entry.
```

Comment:
Can you add some kind of reference to the clearRouteCache() function in the doc comments for the route() call?